clash-lib-hedgehog/src/Clash/Hedgehog/Unique.hs

```diff
@@ -18,6 +18,16 @@ module Clash.Hedgehog.Unique
   , sampleUniqMapBiased
   ) where
 
+#if MIN_VERSION_ghc(9,8,4)
+#define UNIQUE_IS_WORD64
+#endif
+
+#if MIN_VERSION_ghc(9,6,7)
+#if !MIN_VERSION_ghc(9,8,0)
+#define UNIQUE_IS_WORD64
+#endif
+#endif
```

Comment:
Can this be deduplicated?

Reply:
```haskell
#if MIN_VERSION_ghc(9,8,4) || (MIN_VERSION_ghc(9,6,7) && !MIN_VERSION_ghc(9,8,0))
#define UNIQUE_IS_WORD64
#endif
```

I think. The preprocessor language can do C integer arithmetic,

Reply:
I thought I tried that, I'll try again

Reply:
If I feed this to GCC:
```c
#if A || (B && !C)
Yes
#else
No
#endif
```

it correctly obeys the booleans when preprocessing.